client/app/hosting/hosting.service.js

```diff
@@ -1,575 +1,646 @@
+             */
+            resetDomains () {
+                this.$rootScope.$broadcast(this.events.tabDomainsRefresh);
+            }
 
-    this.resetCrons = function () {
-        $rootScope.$broadcast(this.events.tabCronsRefresh);
-    };
+            /**
+             * Broadcast reset cron event
+             */
+            resetCrons () {
+                this.$rootScope.$broadcast(this.events.tabCronsRefresh);
+            }
 
-    /* -------------------------MODELS-------------------------*/
+            /* -------------------------MODELS-------------------------*/
 
-    this.getModels = function () {
-        return OvhHttp.get("/hosting/web.json", {
-            rootPath: "apiv6",
-            cache: "HOSTING_WEB_MODELS"
-        });
-    };
+            /**
+             * Get models
+             */
+            getModels () {
+                return this.OvhHttp.get("/hosting/web.json", {
+                    rootPath: "apiv6",
+                    cache: "HOSTING_WEB_MODELS"
+                });
+            }
 
-    /* -------------------------TOOLS-------------------------*/
+            /* -------------------------TOOLS-------------------------*/
+
+            /**
+             * Get password conditions object
+             * @param {{min: *, max: *}|null} customConditions
+             * @returns {{min: *, max: *}}
+             */
+            static getPasswordConditions (customConditions = undefined) {
+                const min = _.get(customConditions, "min", passwordConditions.min);
+                const max = _.get(customConditions, "max", passwordConditions.max);
+                return { min, max };
+            }
 
-    this.getPasswordConditions = function (customConditions) {
-        const min = _.get(customConditions, "min", passwordConditions.min);
-        const max = _.get(customConditions, "max", passwordConditions.max);
-        return {
-            min,
-            max
-        };
-    };
+            /**
+             * Is hosting performance offer
+             * @param {string} offer
+             * @returns {boolean}
+             */
+            static isPerfOffer (offer) {
+                return /^perf.+$/.test(offer);
+            }
 
-    this.isPerfOffer = (offer) => /^perf.+$/.test(offer);
+            /**
+             * Is password valid
+             * @param {string} password
+             * @param {{min: *, max: *}|null} customConditions
+             * @returns {boolean}
+             */
+            static isPasswordValid (password, customConditions = undefined) {
+                const min = _.get(customConditions, "min", passwordConditions.min);
+                const max = _.get(customConditions, "max", passwordConditions.max);
+                return !!(password && password.length >= min && password.length <= max && password.match(/.*[0-9].*/) && password.match(/.*[a-z].*/) && password.match(/.*[A-Z].*/) && password.match(/^[a-zA-Z0-9]+$/));
```

Comment:
Could be simplified

Reply:
We'll do that in future refactoring task